src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -1604,6 +1604,10 @@ internal bool ReportExplicitUseOfReservedAttributes(in DecodeWellKnownAttributeA
                 reportExplicitUseOfReservedAttribute(attribute, arguments, AttributeDescription.RefSafetyRulesAttribute))
             {
             }
+            else if (attribute.IsTargetAttribute(AttributeDescription.ExtensionErasureAttribute))
+            {
+                diagnostics.Add(ErrorCode.ERR_ExplicitExtensionErasureAttr, arguments.AttributeSyntaxOpt.Location);
```

Comment:
It looks like we can reuse `ERR_ExplicitReservedAttr` for this (via `reportExplicitUseOfReservedAttribute`) instead of introducing new error.